### DISCUSSION TITLE
How to debug QuarkusIntegrationTest in IntelliJ ?

### DISCUSSION THREAD

**Discussion by kirankolli-addepar** [quarkusio/quarkus#33421] [Q&A]
When I add breakpoints on application code and then run my Quarkus Integration Test ( via quarkusIntTest gradle task) the code execution never pauses on the breakpoints.  
Any solutions to this ?

Quarkus version = 2.9.2Final
IntelliJ version = IntelliJ Ultimate 2023

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
Assuming you are testing a jar, you can set `quarkus.test.arg-line=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005`

**Reply by kirankolli-addepar** [reply]
do you mean override `quarkus.test.arg-line` property by passing a VM args via `-Dquarkus.test.arg-line=...` ? If yes, that did not work.

**Reply by kirankolli-addepar** [reply]
to clarify the command used to run the test by IntelliJ is this : 
`gradle quarkusIntTest --tests "<Class name>" -Dquarkus.test.arg-line=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005'` and this did not work

**Reply by kirankolli-addepar** [reply]
**Next update on the answer :** 

Add a system property using the values as mentioned like so : 
![image](https://github.com/quarkusio/quarkus/assets/122354105/d4074f1f-31d8-4937-ab2b-79f6b079f731)
Also create a remote debug config for your application on IntelliJ to connect on localhost:5005 like so : 
![image](https://github.com/quarkusio/quarkus/assets/122354105/82f915b5-9f74-4944-82c0-e6f2a9a9a589)
 
If your integration test used RestAssured to test a rest endpoint make sure to increase the timeout ( see here: https://github.com/rest-assured/rest-assured/issues/1444) 
Then start your gradle quarkusIntTest task and then also start your remote debugging session and this should work.

I wish it was easier than this though.

**Reply by maxandersen** [reply]
Tip is to set suspend=y so it Will wait for Connect. 

Also using "Attach" from intellij or eclipse removes need to setup the debug config.

**Comment by kirankolli-addepar** [comment]
Are you suggesting to start the test as so ? 
![image](https://github.com/quarkusio/quarkus/assets/122354105/e769bc6b-72d3-4a92-b13a-e90ae78fcb13)

If yes, this option does not work. The test fails with the above setup. 

And without `quarkus.test.arg-line` it never stops at the breakpoints. Usually I had never had to deep dive so much just to debug with other frameworks. With quarkus its unusually hard and totally not dev friendly which is the USP of quarkus being touted around.

**Comment by geoand** [comment]
Using `quarkus.test.arg-line=-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005` in `application.properties` does work when you run `mvn verify`. The integration test waits until you attach a debugger

**Reply by kirankolli-addepar** [reply]
but this still requires a separate debugger to be attached which is not a good developer experience at all.

**Reply by geoand** [reply]
Indeed. I suggest opening an issue on Jetbrains issue tracker

**Reply by kirankolli-addepar** [reply]
isnt this a quarkus problem though? the way integration tests have been implemented necessitates this.

**Reply by geoand** [reply]
It **is** intended that  `QuarkusIntegrationTest` launches a **separate** JVM, it's not a problem/bug.

**Comment by kirankolli-addepar** [comment]
Yep tried this too, but it didnt work.

**Reply by ennishol** [reply]
I tried to put `quarkus.test.arg-line=-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:50` into application.properties and ran mvn verify and it did work. I could debug it by attaching to process with IDE
